csrc/scheduler/normalization_inner_outer.cpp

```diff
@@ -394,12 +408,20 @@ std::pair<int64_t, int64_t> getBufferBatchSizeAndThreadsPerBlock(
   // reaches batch_min.
   int64_t threads_per_block = threads_per_block_min;
   int64_t inner_batch = ceilDiv(after_vectorization, threads_per_block);
-  while (inner_batch > batch_max &&
-         threads_per_block + warp_size <= threads_per_block_max &&
-         ceilDiv(after_vectorization, threads_per_block + warp_size) >=
-             batch_min) {
-    threads_per_block += warp_size;
-    inner_batch = ceilDiv(after_vectorization, threads_per_block);
+  if (shared_memory_buffer_size > 0) {
+    while (inner_batch > batch_max &&
+           threads_per_block * 2 <= threads_per_block_max &&
+           ceilDiv(after_vectorization, threads_per_block * 2) >= batch_min) {
+      threads_per_block *= 2;
+      inner_batch = ceilDiv(after_vectorization, threads_per_block);
+    }
+  }else{
```

Comment:
didn't change heuristics for cases without smem persistent, leave it for following heuristics tunning works.